components/ecma402/src/pluralrules.rs

```diff
@@ -0,0 +1,344 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE )
+
+//! Implements the traits found in [ecma402_traits::pluralrules].
+
+use icu::plurals as ipr;
+use icu::plurals::PluralRulesError;
+
+pub(crate) mod internal {
+    use ecma402_traits::pluralrules::options::Type;
+    use ecma402_traits::pluralrules::Options;
+    use icu::plurals::{PluralCategory, PluralOperands, PluralRuleType};
+    use std::cmp::{max, min};
+
+    // The difference between `first` and `second`, clamped on the bottom by zero.
+    fn clamp_diff(first: usize, second: usize) -> usize {
+        match first > second {
+            true => first - second,
+            false => 0,
+        }
+    }
+
+    /// Converts the trait style option to an equivalent ICU4X type.
+    pub fn to_icu4x_type(style: &Type) -> PluralRuleType {
+        match style {
+            Type::Ordinal => PluralRuleType::Ordinal,
+            Type::Cardinal => PluralRuleType::Cardinal,
+        }
+    }
+
+    /// Formats `n` as a string representation of fixed decimal, based on the supplied options.
+    pub fn fixed_format(n: f64, opts: &Options) -> String {
```

Comment:
I'm a bit confused. Shouldn't that be handled by fixed-decimal? If not, should we abstract it, because the same operation (take `f64` and `NumberFormat::options` and produce a string) will be used in other formatters?

Reply:
I wanted to decouple implementing ecma402 traits from the discussion on how to construct `FixedDecimal`.  It would seem reasonable to pull this functionality into `FixedDecimal` as a separate step, provided that you all find it to be reasonable.  This could be pulled into `From<(f64, NumberFormat::options)> for FixedDecimal` or some such.

Reply:
Can you file an issue about it reference it here in a comment?

Reply:
Done.